components/brave_rewards/browser/rewards_service_impl.cc

```diff
@@ -3497,6 +3507,20 @@ void RewardsServiceImpl::SetAdsEnabled(const bool is_enabled) {
       AsWeakPtr()));
 }
 
+bool RewardsServiceImpl::IsRewardsEnabled() const {
+  if (profile_->GetPrefs()->GetBoolean(prefs::kEnabled))
+    return true;
+
+  if (profile_->GetPrefs()->GetBoolean(prefs::kAutoContributeEnabled))
+    return true;
+
+  auto* ads_service = brave_ads::AdsServiceFactory::GetForProfile(profile_);
```

Comment:
This is definitely not the right thing to do here and the previous `IsRewardsEnabled` was intentionally removed. We should always be using `ads::prefs::kEnabled` for this checks and really this method should go away and `prefs::kEnabled` should be set to true/false based on the value of the other prefs as needed. cc @tmancey because `ads_service->IsEnabled()` should also be removed

Reply:
this is also the root cause of a crash in the tests because the timing of it triggers AdsService to be added twice to the dependency graph

Reply:
https://github.com/brave/brave-browser/issues/15434